/*
  --- Day 9: Stream Processing ---

  A large stream blocks your path. According to the locals, it's not safe to cross the stream at the moment because it's full of garbage. You look down at the stream; rather than water, you discover that it's a stream of characters.

  You sit for a while and record part of the stream (your puzzle input). The characters represent groups - sequences that begin with { and end with }. Within a group, there are zero or more other things, separated by commas: either another group or garbage. Since groups can contain other groups, a } only closes the most-recently-opened unclosed group - that is, they are nestable. Your puzzle input represents a single, large group which itself contains many smaller ones.

  Sometimes, instead of a group, you will find garbage. Garbage begins with < and ends with >. Between those angle brackets, almost any character can appear, including { and }. Within garbage, < has no special meaning.

  In a futile attempt to clean up the garbage, some program has canceled some of the characters within it using !: inside garbage, any character that comes after ! should be ignored, including <, >, and even another !.

  You don't see any characters that deviate from these rules. Outside garbage, you only find well-formed groups, and garbage always terminates according to the rules above.

  Here are some self-contained pieces of garbage:

  <>, empty garbage.
  <random characters>, garbage containing random characters.
  <<<<>, because the extra < are ignored.
  <{!>}>, because the first > is canceled.
  <!!>, because the second ! is canceled, allowing the > to terminate the garbage.
  <!!!>>, because the second ! and the first > are canceled.
  <{o"i!a,<{i<a>, which ends at the first >.
  Here are some examples of whole streams and the number of groups they contain:

  {}, 1 group.
  {{{}}}, 3 groups.
  {{},{}}, also 3 groups.
  {{{},{},{{}}}}, 6 groups.
  {<{},{},{{}}>}, 1 group (which itself contains garbage).
  {<a>,<a>,<a>,<a>}, 1 group.
  {{<a>},{<a>},{<a>},{<a>}}, 5 groups.
  {{<!>},{<!>},{<!>},{<a>}}, 2 groups (since all but the last > are canceled).
  Your goal is to find the total score for all groups in your input. Each group is assigned a score which is one more than the score of the group that immediately contains it. (The outermost group gets a score of 1.)

  {}, score of 1.
  {{{}}}, score of 1 + 2 + 3 = 6.
  {{},{}}, score of 1 + 2 + 2 = 5.
  {{{},{},{{}}}}, score of 1 + 2 + 3 + 3 + 3 + 4 = 16.
  {<a>,<a>,<a>,<a>}, score of 1.
  {{<ab>},{<ab>},{<ab>},{<ab>}}, score of 1 + 2 + 2 + 2 + 2 = 9.
  {{<!!>},{<!!>},{<!!>},{<!!>}}, score of 1 + 2 + 2 + 2 + 2 = 9.
  {{<a!>},{<a!>},{<a!>},{<ab>}}, score of 1 + 2 = 3.
  What is the total score for all groups in your input?

  --- Part Two ---

  Now, you're ready to remove the garbage.

  To prove you've removed it, you need to count all of the characters within the garbage. The leading and trailing < and > don't count, nor do any canceled characters or the ! doing the canceling.

  <>, 0 characters.
  <random characters>, 17 characters.
  <<<<>, 3 characters.
  <{!>}>, 2 characters.
  <!!>, 0 characters.
  <!!!>>, 0 characters.
  <{o"i!a,<{i<a>, 10 characters.
  How many non-canceled characters are within the garbage in your puzzle input?



 */
const assert = require('assert')

class StreamProcessing {
  constructor() {}
  groups_score(input) {
    // 1. loop with the input stream
    // 2. identify garbage
    // 3. identify groups, with score, might be need a stack here
    // 4. sum the groups score

    let sum = 0
    // stack to trace '{', for score
    const stack = []

    let score = 0, garbage = false
    for (let i = 0; i < input.length; ++i) {

      switch(input[i]) {
        // NOT garbage, push '{' to stack, score++
        case '{':
          if (garbage === false) {
            stack.push('{')
            score++
          }
          break
        case '}':
          // NOT garbage, pop stack, sum now, and score--
          if (garbage === false) {
            stack.pop()
            sum += score
            score--
          }
          break
        case '!':
          // skip the following one character
          i++
          break
        case '<':
          // start garbage period
          garbage = true
          break
        case '>':
          // end of garbage period
          garbage = false
          break
        case ',':
          // reset score, if has ',' AND stack is empty
          if (stack.length === 0) {
            score = 0
          }
          break
        default:

      }
    }
    return sum
  }
  garbage_count(input) {
    // 1. loop with the input stream
    // 2. identify garbage
    // 3. identify groups, with score, need a stack here
    // 4. sum the groups score

    let sum = 0
    // stack to trace '{', for score
    const stack = []

    let score = 0, garbage = false, garbage_char_count = 0
    for (let i = 0; i < input.length; ++i) {
      switch(input[i]) {
        case '{':
          if (garbage === false) {
            stack.push('{')
            score++
          } else {
            // during the garbage period, count++
            garbage_char_count++
          }
          break
        case '}':
          if (garbage === false) {
            stack.pop()
            sum += score
            score--
          } else {
            // during the garbage period, count++
            garbage_char_count++
          }
          break
        case '!':
          // skip the following one character
          i++
          break
        case '<':
          // during the garbage period, count++; otherwise start garbage period
          if (garbage === true) {
            garbage_char_count++
          } else {
            garbage = true
          }
          break
        case '>':
          // end of garbage period
          garbage = false
          break
        case ',':
          // has ',' and stack is empty, reset score
          if (stack.length === 0) {
            score = 0
          }
          // however, during the garbage period, count++
          if (garbage === true) {
            garbage_char_count++
          }
          break
        default:
          // during garbage period, count++
          if (garbage === true) {
            garbage_char_count++
          }
      }
    }
    return garbage_char_count
  }
}

module.exports = {
  StreamProcessing
}

if (require.main === module) {
  const stream = new StreamProcessing()

  let input = `{},{{{}}},{<a>,<a>,<a>,<a>},{{<ab>},{<ab>},{<ab>},{<ab>}},{{<!!>},{<!!>},{<!!>},{<!!>}},{{<a!>},{<a!>},{<a!>},{<ab>}}`
  let output = stream.groups_score(input)
  console.log(output)
  assert.equal(output, 29)

  input = `<random characters>,<{!>}>,<{!>}>,<!!>,<!!!>>,<{o"i!a,<{i<a>`
  output = stream.garbage_count(input)
  console.log(output)
  assert.equal(output, 31)

  console.log('======')
  input = `{{{{{{<!>i!}iu>}}},{{{<}!>,<eoo{u!!!>eae{e>}},{{{<,,a!!}{!>},<!,ai!!<!e>},<{i{,{!>},<>},{{<!!aia'!>i!>},<u>},{{<'!>},<i!>,<!>,<o>},{{{{{},{<ei!>},<u}!>,<o}!>a!>,<!!!>},<>}},{<e","}o}!>},<u>}},<}oae!!!>{,!o}!!!!o!!!>},<oa!!>}}},{{<!>!!!>a>}}},{<!!uae}!>},<!>,!!!>u!>e}i{!!!{ui!!!>>,{<!e}!>i<e!!!>!>,!!!>!>},<!!o"i}'!>},<!!>}}},{{}}}},{{{<!>,<{!!o!!!>},<!>},<!!!>{!>},<!>,<!!i>,<!!{a<i!!!>!!!>>},{{<!>!!i'>}},{<!!!>"!!!>!!!>!>},<!>},<ae,>,{<!u!!!>"!!u!>,<>}}},{{{{<"!>},<o!>},<!>},<oo!>>}}},{{<!>},<a'!!u'!!<!!">},{<<o'!!!!!>>,{{<u"i,{ai>},{<,"!!,!>},<{!>o!>ii!!<ii!i!>,<e'>}}},{{{<!!!>,<,'{!!!!!!}>},{<}a!>,<!u"a!!{oa"i>,{<'}}"<"<{,<a>}}},{<a{,i"e!!u>}}}},{{{{}},{{{{<>},{{<!!!>!!eu<<!!"u!>},<!>},<o!!!>},<>},<!!!>,<',o!>},<!>},<!>},<'>}}},{}},{{},{<}!>,<!!oi!>},<!>,<'"!>,<!!!!}>}}},{{<<'!!o!!e!>!!!>ee!>,<!>,<{>}}}},{{{{{{<,!o!!<a'!>},<!!!>aoo!>,<aei'!>>},{<>}},{{<!!!>}!>,<!!"ea!!!>},<}"e>},<i}!!!>o>}},{{{<i!a<!>!!!>'!!!><!!i!!!!i>,{<{}!>,<u{!!o!>,<a{!>},<!}>}},{{<"e!>},<<>}},{{{<!!!!!!e!!!><"!!!o!!!oi!!}>},<i!>,<!!!>i',},!>,<!>},<o!!!>,<e}>},{<a{!>!>!o}oo'u!!}i!>!>iu!">}}},{{<'"!}!!a"'!!!>,<!>},<>,{}}},{},{}},{{{},{{<!>},<{u!>,<!!!>u"!!!!!>!!uu'!!!>!>},<,{>}}},{{{{<!!i!>,<!>e!!!>{i!!!>!!!>o!>!>},<oe!{!>,<!o!!!!!>>,{}}}},{<}o<oi{'<!!<{>}}}},{{{<i}!>!!oe!!o>}},{<!!!a!!}!>,<">}},{{{<!!<!>,<!>''}>,{<o>,{{<u{!!!>{{"'">},{<a!>},<'i!>,<>}}}},{{<{ioe}!>!>!!{!e{}!e<!'!!{>}},{{<!!!>,<a!'<o!!!!a!!!>!!a"!!',}">,{}}}}}},{{<"o!>,<e!!!>,<{!>,<'!>},<u!>!!!>!!!>'!!!!"!!!>>}}}},{{{{{{<!!!>},<!!!>>}},{{<a!>,<au'!!!><"!!a!{eiuu{,>},<<o!>"!>o>},{<>,<o>}},{},{{<}!'!>!>>}}},{{{<!>},<!,!}!!!!!>,<!>,<!!a'!!!!!>,!>'}!!>,<!>,<!!!{!'!>!>},<!!!!!>>}},{{{{{<"u!!!>"!!'e!>!!!>},<,e!>}''!!!!{o!,>}},<!>,<!u>},{<,!!u!>,<!!!>a!>},<,!>},<<i'!"!'i!!!>>,{{<!!'}!!{o{o">}}}},{},{<"!!a,!>,<!!<ai!!!!!>"a!>oe{!!!>!i<>}},{{{{<<'!!!>"!>},<{!!!!'i>},{<"!>!!!>ee!!!>},<{!>},<ei!>},<!!!!!>!!!>u!>!!'i>}}},{{{<!!!>!>,<!>!i!!!>!!!>>},{}}},{{}}},{{{{<'!!!>{i<!!!!eee!>!>">},<}!!!!!>{!}u{!!!!!>!!!!!>},<<<!!>},{<!>,<{!>},<!>,<"o!!}<!!!!!!!!!!!>!>,<!!!>>,{<!!!>,<!!<u"e>}}},{{{{{{},{<!!!<u!!i!!<"!>,<'>}},{{{<{!>,<!!!>},o{o'u!,e!><a,,!!!>,<>}}}},{{<e!"!!!!!>,<}'<,!>,<!!a}!!}{}>}}},{}},{{{{}}},{},{{{<a>},<{!!!>,!!!!!!{o'!!!>},<e!>}!!!>{u>},{<!!{!!{u}!!!>'!!!>!>},<!o>,<!!!>!!!>!!!!!><,!!!!!>},<!oi!>>}}},{{{{{<}!!!>{ui,!!!!"!{>}},{{<!!!>!!oaoi!!!>,<!!<!>,<{>,{}},{{{<}!>},<''u!u!>!!!>>,{<>}},{<!>,<!!!!{!>eeu!!a!>},<!>,<!!o!!!!a>}},{<>}},{{},{<'!!!>oa!>,<e!>,<e!!!>!!'!>},<>}}}},{{<!!<!"i{}>},{}},{{},{<a!>},<}<!!!!'}!>},<{!>,<!!a>},{{<'>},{<e!>},<!>},<>}}}},{<,"!!!>,!"e!!'''!!,!>,<!!o!!u>,{}},{{<!{oo}!!o!>},<!>,<'o}{>},{{<u!>''>}}}}},{{{<>},{{<u!>,<i!!!!!>}}!!!>},<!>>}}},{{{{<!!ao{ou},>}},{{<e"},}!!!>i'!>,<!'{!!i>},{<<!>,<!>},<u!"i!>{,!u!!<'a'>}}},{<!!!>!!!>'}!>},<>,<!!!>!>,<">},{{<!!}a<u<a!>eu{}<<!>,<io>},{{}}}}},{{<au!,,'!a,,!!!!!>!>'!>}}"!}!!ia<>,<}!>eoo>}}}},{{{{},{{{<}e!!!!o!>,<i!>,<!{!>},<!!e!!"!>},<>,{{<!!!><!!<i!!<e'!i""!!aa!!""!!>}}}},{<o,o}uu}u,}!!a!!a,!',ie>,<!!!>!>},<}!>!>},<!!!>oo<<!>eau!o!!>},{{<e<{{!}}!!!!}}!!}>,<,!!!>,<!a,'ae!!!>!,!>,<"{e!!!>oo>}}},{{<}!>a<<>,{}}}},{{{}},{{{{{{{<e!!!>},<!!<!!",}!!{",>}},{<!>,<!!!>!!'!>!>!"'{!>},<}!>},<''">}},{{<!>},<>},<!!!>,<!>,<!>},<}}uiu!!i!>},<i!!>}},{{{{{<!>}!!{!!!>'!!!>!!!>!!!>!!!>!{aei!!">},{<!>!e!>},<!<>}},{{{}},{{},<o>},{{},{}}},{{<e!!!>u!!o!!!!i<o!>},<e<!>,<!!!>>},<!>!!'!<!,a"<!>},<!!!>!!a!>,<<}>}},{<o!!e>,{<!>},<!><!>},<!!>}},{{},{},{{{<o!>a!au!}!!!>!>},<eu!>},<!>,<,!oe,!>},<}>}},{<a{!!ui!!!!',<a!>}>}}}},{<!!!!!>{!>},<!!eo!!!>i!{i!>'i}!>},<!!!>>}},{{<}"!>,<!,!>,}!!!}"!>!!!>!!',!!!!'>},<u!!{!!a},!!!>>}},{{<{!!!>o>},{<!e}o!!!}!>,<io",>}},{<{{>,<ie'!!!!>}},{{{<!>!>,<!<o!>>},{<!>,<',"e!!!>!<!>,<'>}},{<""!>},<a,,uu>,{<u!!!>!!au!!!>}!oa!>},<{!!!>!!!!a!!!><e,'>}},{}},{{<u!!a!>,<!}!!!!!>eioae{!,>,{}},{{{{},{{}}},{{<oi!>,<!!}!>,<>},<!!e!!!!!!!>!>},<'!>},<{!u!!{!{aa!>,<>},{<}>}}},{{{{{<oeuo!!!>!>},<{'!!!>}>}},{{<{>},<i!>},<{}o>},{<!!!>{!!!!'!!o!!<!!a!!!!!o"!<!!!!}!!,e>,<aea>}},{{{<>},{}}},{{{},{<!!!>o!!!!}"!!!>!!!!<e!!,!!!>!>,<eu!!!>},<!!iie>}},{},{{<e!!>},{<!'},{,!!!>,<!!!>,<!>},<'!!!>,<>,{<e"!>},<{!>},<"aa!!<!!!>"!!>}},{{},{{<'!>'!ao!!!>a{""!><!>>,<a!!!!!!e!!{ia<!{!!e'!>,<,!!!>!>},<oa!!<>},{<ae,o'!>,<e!,e{>,{<!!{i"a!>,ui}>}},{<<e>}},{<!!!>,<!'!>!!!>>,{}}}}}}}},{{}}},{{<'!!!!o!>,<!>o!!!>!e!!<!!'o!a!!>},{{<!>!!!>},<!>ae"'!!<!!a!u!e>,{{<!>,<{,>},{}}},{<!!!!i!!o!>,<<!>a!!!!!>!!!<>}},{{{<,!>,<!!{!!<i!!i'!}!!"!!!,}!>},<!'!!}e!!ua>}},<o!>{{{!>},<,!!u>}},{{{<!>,<!!a!!!>"{i}!>},<e}u!!!>},<}!e>},<i'!!ee'!!}>},{{<!<iu!!}!!!!<e!!i>}}}}},{{{{{{},{<uu!>'ae>}},<!!!>e}!!!>"a'o}!!!>{}!!uaei>},{{},{<!>!!!>o!}i"!!}>}}},{{{{}}},{{<'!!'!!e'>,<ei!!!>'!!!>,<ii}<!>,<!!!!!>o>}}},{{{{<{!>},<!>}!!>,{<<!>,<",}!!u!>"{"e!>,<'!>!>!!!>i!>},<>}}},{{{{}}}},{<<}i!>,<}!>},<'o!!u"{!!!>,<!>,<o"!,i">,{{<!!!>},<i,!!i!>!>},<!!!!,o{!!!>},<!!i!>,<"o'!>>},{<u!!!!!<{u!oi!!!>},<}!>e!>},<>,{<!>},<!>!>,<{!>,<!!!>}"<eoo!!a!!!>!!<">}}}}},{{{{<,!!{!>,<<uai}!!!"}>},<oi}!"!>,<!>>},{<!>},<}!!!>a>,{{},<!{o!"}!!}!!!>!!!!e>}}}}}},{{<!>!i>},{{}},{{<<!>e!e'!!,!>},<{u!>},<u!a>},<{!!!!}{!!a>}},{{{}},{{{<!!!>!!!>!!'"!!!!!!}a>}}}},{{{<,!!!!!>,}!e!>,!>!!e>}},{{{<!!!>i!!u!!<{aee!!!a!>>},{<e!!}a!!"<!'!!<a!>u<ia!>!!!>}>}},{{}},{{{},{}}}},{{<!>!ea!!!>u!!o!"o!<'>,{}},<!!!!!>},<uuoaa>}}},{{{},{{<!>},<!>},<o!>},<i!"!!o!!>},<,i}!>o!>},<>},{{{<u}!>},<ii!!!>'>}},{{<>,{<!!'!!"!!!>!!},!!u!u{!>!>},<!<}!'!>!!!!">}},{{<o!!<"e'!o!!,,!>},<>,<<a!!{>},<uu!!o!!!>,<!>,<{!>!!!!"!!,!!!>u!>},<u>},{{{{<>}},{{<"'!!!>a""{a!!<!>,<!>{{>},<<a!>,<"u!>,<"!,,e!>,<!!}{!!!>>}},{{{{<">},{<"<ii,{!!!>!!!>!!<!!i''!a!>,!!!>>}},<'!>},<!>},<a{>}},{{},<ie!>},<a}!>,<iio!>,<>}}},{}}},{{{{<,!>},<!!"!!ou{ai!>},<i}!!!!!>,<"!><<!">}},{<>}}},{{{<!!!>!>},<!!!>!>,<!,io!>,<!!!>,>},{{<"!u{!ia!>,<a>},{}}},{{},<!>},<u{,!!!!!!'!!"e!!!>e!>e!!!>ua!o!>'>},{{{{}},{{{},{}}},{}},{{{<!>},<!!!>a!!!>,!>},<!!!>},<!>},<'o!!eo>,{<,u<'!!oa!!!>!,e!>,<!!!>!>,<o!i!<!!{>}}},{}},{{<!!},>},{{{<<>}}},{{<>}}}}}},{{{<,>,{{}}}},{{{<!>},<!!!>a!!,'io!!!!}uo<'!!,"}>},{{{<i,,aoai!!!>!!!>,<!!}ei!!!e!!u!!!>},<>},{<!!<!>,<eo!!!!e!"{}eu!>'i<!!!>},<>}},{<>},{}},{<!!!>!!!!!!oia!>},<iio,{!>!!aa!!e!!!>{!!!!!>>,<>}},{{<"oi!!eoe{{'!}!>!'',u>},<iuoua>}}}},{}},{{{},{{},{<!>,<o!!,!!o{!!!>"!!!>!!!>!!!>,<!!!!!>u!>},<{i>}}},{{<!!!>{e,!!!>}e!o"o}!">},{{<<!!!!!>!>"!eau!!!>"ou"!!!!i!!e!!!>>,{{},{<!!!>u{!>!>,<'!!!>ue>}}}},{{<'i}o"'eoe>,<!!a!!e!!!>}!!e"'>}}}},{{{{{{{<oe{!>,<!>,<!!>},<!!!!,o<<!>},<''!!!>!!!>!!}!!"!>,<!>},<oi<>},{<!!!!a!!!>,!!e!!i,'>,{<a'!>!>'!!'i"i{",>}}}},{{{{<ue"!!!>},<!>},<{i"!'!>,<>},<o!!}!!,,!!!>!!e'<i">},{}},{{{{{{{{<<}{!!!>aoo!!<!!<!>!,i',!!<!!!!!!'>,{<<!>},<'!!}"!!{!!!!o>}}}},{<!>!>,<a!i!!!!',!!a!oi!!e>}},{<i{!>,<!!!>,<!!!>},<a!i!!eo!!"'!>u>}},{{<!>},<""!>},<!>,<!!!>!>u!!!>>},<!!!>>}},{<>,{<i>,<!>,<!>!!!!i'!!!>u!>,<<<}!>},<e<}!>},<e>}},{<!{,ai!<aio!>},<!ue!!!!o>,{{<!>,<o!!!>u!!!><eu<!!o}i!>,<>},{<!!,'!>,<!!<!>,<!,>,<!!!>,<e!!!>>}}}},{{<i!!!>!>},<'i!,<<{"!!!!{<>},{}},{{<o{!>},<'{!>o'}>}}}},{{<}}>},<"uiu>}}},{{{{{<!>i!!""!!}!>'!!!>}{ae}"!>,<e<>,<!!''!'!>,<'e,!>">}}},{{{<{!!!>>,<e"ae>},{{<o!!!,<u!>},<}"'!!oi!ei,!>!>,<i!!u<>},<!{!>},<}!!u"u!>}!!<>},{{<"'",,!!!>!!o"!!!>>,<!{!>},<i!>},<,!!u<<>},{<oi!>,<i"!>'!,'}}a!!,',e'!>>}}}}},{<!!'ia!!!o!!!>u!>,<,i!!'!>!>},<!>},<!!!}<!!!!!>!>>,<'a"!!!'o!>},<!>{{o!>,<}a'>},{{{<o'!!!>i>},<>},{{},<!!!>,!!"!o!>},<!>!>!oo>},{{{<!!o!>,<a!!!>!!!>!!!!">}},{{{}},{{<"e}e!>},<!>!>oaa!><!!i!!!>i{i}>}},{}},{{<'!!,a!>i!!!>},<>},<>}}}},{{{<o{i,}!>"!!,!}auo!>,<"!!>,{{<o!"{i<!>},<>}}},{<>},{<{!!!>e<e'oa!!"!!!!!!!>>,<ieo!!!><u!!,"!>},<!i!!!>i!>},<o,!!!>e!>,<>}},{{},{{<!i{>},<iu!>!!>},{<'e",}{!!a!>!!!o!>}!!>,{}}},{{<<>},{<>}}}},{{{{{<!!!>,<,!>,<>,<!"!>},<e!!'e'{!><e!>,<a{{o'!!{>},{<e!!!>{ai}{!>},<!!}o!!!!<'!,!!!>},<}!">}},{<o!!oao{!!u<!!}!!<!>i!!a!o>,{<{!>,<>}},{{},<e!!"!>},<!>!!o!!!!!>,{!>},<'",iae<>}},{{{},{<!!!'!>,<!!!>!!<e!>i>}},{{},<!!!!!>},<!>},<!>i"e!!!>u}iiui!>a!>},<o">},{<,{,}<>}},{{<}!{'!!!!a!!!!!>,<e!o!!'"!>,<!!!>>,{}},{<<ei{!!!>!!!!}!!!>!>!!e}a!>},<!!!!}>,<ii!!eo},!>ue,!"""i}i!>o>}},{{{{<!"","''<e},!>},<>}},{{{}}}},{{<!!!>!>},<}{u<a!!,"i!!<>},{<!!ioa,!!!>!!uo!!>,{}},{<i!!>}},{{<<!!!!!!!>!!!!'!e!>,<e'!>},<">,{<i!!a!>,u!>},<!>,<!>},<!{'>}},{},{{<!!a!!'!!!>,<>},{<o<u}!>!>e"u}i!!!>>}}},{{<{!!{ei'!"!,e"!>},<>},{{{<,,!>},<a>,{<>}},{<i!!!<<>,{}}}}}}},{{{},{<o}ou!>,<!!!!!>!>,a}ue>}},{{},{<i!>!>},<i!>ue!>'!!!>!!!>'!>},<!>!i!ao!>},<!>>}},{{{<e!!a!!}!>,<"'!>},<e{,,!!!>{!!!>},<{!>,<!!>}},<{!!!!!>u!>,<!i!!!>!!!>'>}},{},{{{{<u}>,{}},<oi!!!>!>,<>},{<!>{!!u{!,!!o!>},<!>},<u>,{{{<!!{}eu}!!!!u!!!>!>},<!!!>e!!"!>!!!>!aea>}}}},{{{{},<u"u<<!!{!>},<!u!!}!!!}!i<u'!>,<!>,<>}}}},{{{<u{!!'},!!!!ui>}}}}}},{{{{{<>},{{{<!>},<e!!!!!!!>!!!"e>}},{{<i}{!!!>"e">}},{}},{{<!!"!!!>!!!!!!!>!>},<!!!>i!!<e!>,<!!!>!>},<!!!>},<>},{<!>},<{!!!>!>,<!!a!>!>!!u!!!!ue'"e{i'>,{<!>!>u{!!!>,<"}o<!">}}}},{{{<!{!><>,{{<ii!>},<,i'!!}e}>}}},<!!!e!!{a>},{<!'!!!>,<o"aa>,{{}}},{<i!<u"u!>!!a!>},<>,{}}}}},{{{{{<'>},{<!>},<!!!>!!!!u!{>}},{},{{},{{},{<{!!!>!!{}!!<!"!,!!!>!!u,!>,>,<>}},{}}},{{{<o,!>,<!>},<!!o!>},<}a!{"{e>}},{{{<a>},<uue!>!!o}!>,<!>},<>},{<<!!i",ae!>!'!>},<!ei{a{e!!!>!!!>!!"!>,<>}}},{{{{{{<"!>}>}},<>},{<>}}},{{<"!!!>ei!a,u>},{<!>},<!a{,<!!!>}{!!>,{}}},{{<a}!>},<!><!!a!!!!!!"'!>},<>,<a}<!>,<iu!u!>,<eo>},{{},{{},<!!!u!!!!>},{<ue>}},{}},{{{<e}!>},<!!!>o!!!>!>,<"o!!!!<!>,<i}'!!!>},<">}}}}},{{{{<>,<!}!'!!e!!!}}a!!<>}},{{{<!!!>>},{{<!!'u!!!!!!!!oi!>},<!!!>uia!!!>!>>}},{<}!!'!!}"u!!!>o!!!>!>},<>,{{}}}}},{{{},<!!!>},<!!!>{o!!}!!!>">},{{},<!!<>}}},{{{{{},{<a}!!uo'<!>i,,">}}},{{{<aa'!>e!>a,>}}},{{<eo!!!>'"o!{!!!>}"!>},<oe!!!>,<!>>,{<}!!!!'!>,<>,<!!!!!<"!>!>,<>}},{{<"oi!>,<!>!>ei!!!!!>o"!!!>!!!!>},<{'!!u!!!>}i!!!>,!>a}"}a<!>'i>}}},{{{{},{{{<!!!>!!"i!!!>!>>},{<ui}>}},{<!e!>,<<!>i,aa!''">,<io!>!>,<}{a<e{'{!!{!}!!e!!!>!>},<">},{{<!!!>!>,<!>!>},<oi!!!>},<"!>a,!!iu!!>},{<e,!>!!!!{{!>,<e,!ua{"ai!!!>io!<!!>}}},{{<o!>i!>},<{!>!>},<>}}},{{{<a!>,<!>},<'<,o'}}o!>,<!!!>!>,<ae"a">}},{{{<!!!>,<o<!>,<!>,<}}u!!!!<!i!!!!!>,<>},{<!>!!{!!}e!!o"!>,<,o<!!!>,<!!,>}},{{{<}a!>!!!>},<>}},{{{<}!>,<}>,{<!>},<!"!>,<"'!>},<"!!}"!!!>!>!>!!!>{!!>}}},{{<!!!!!>},<{,,<"u!!"o!}!ao!!!!,e!e,ie>}},{<!!!>!>!!!>u"a!}!!a!!"}>,<!>!!{a!!i!!!>a!}{!!>}},{<!>,<!!!!a'!'!}{!!!>>}},{{<!>},<ai>},{{<>}}}}},{{{{<uu!>,<!e!!<ea!!a}{a{u!>},<!!}!!!>>},{{{},<!>,<!<!>,<!>,<{!!!>,<a}"o>},<'!>},<!u!>!>},<u<!!ua!!!>{!>}>}},{{{<e!>!>,<!!<<}!>,<e!>{!!!!{}>}}}},{{{{{<i'i!!!{!!!>,<!!!o'!}!!e<}<>},<>},{}},{{<!!{!!"!!!>},<o!'!>i">},{<!>},<!!!!{!!}!>},<!>},!>,<oe}!>i!>},<>}}},{{{<ea!!ua,!!}!!!>},<!e!!!{!,'>},{<ii{!>,<!><o!!i{!>,<!'i!!!!!e,!'!",}>}}}}},{{<!'{!!!>!!'!!!>,<!!!>},<a>,<!!!>>},{<!!!>au>},{}}},{{{{{},{{{{<a"!>!!!>,{i!!!>ia!>,<u>},<>},{{<}!>},<!}}!>},<>}}}},{{<!>},<!>},<i<!!>},{<,!>,<e>},{<'!!!!!>>}}},{{{},{{<,!,,!,'>}}}}},{{{{<!!u!>},<>}},{<!>},<!i!!!>!!!!"!ue"a<!!'!a!>},<!>,<>,{}},{<!>},<>}},{{{{{<!>},<!!!>,<'!>},<!!e,!>},<oe!!"!!!!,!!ae!!!>},<>}},{}},{{<o>,{<"'!>,<!!!>!!!>,!!!>},<ee!>!!!>!>>}},{{{}}}}},{{{{{},{{<o!>},<!!!>},<'!>},<>},<{!!!>!!aa>}}},{{<!!!>!!!>!>eou!e!>,<}',u!!i"">},{<{!!u!!<!>!>!><!>,<{,>}},{{},<ii!!a}i"!!'!>e}!>,<!!!>,<"!u!>},<}>}},{{},{{{{<}!!u!!i!>,<!!!>}ui,a<!>,o}>,<!!!>!!!>!!o"u!>},<>},{{<!,,"",'u!>,<!<o!>,<ua>},{<!>,<i!>,<!>},<!>,<!!!>},<'!>},<!o,a!'!u,!>},<}>}}},{{<!!!!ee!!"!!!>,!>ee!!!}!!!>!ua{<,!!!>!>!!u>},{}}},{{}},{{<e'},u>},<i!!!!a!>o,u',!!!'{"!>},<oiu>}}},{{{<'i!>},<!!!!!>!>},<o!!i"'!!!!u!!u!!!>}{{!!,'>},{<o"e!>},<{!>}o,!>o'>}},{<i}>,{<e{>}},{<!{>,{<!!u<!!!iea!!!>},<>}}},{{{<o!!!>,!u!!e!!!!}!>},<!>,<i'{!!!>aouo>}},{},{{<!>},<<a','<!!!!!!!e!>,<!}!u!>,<!>u!>e!!u!">}}}}},{{<!!,!>},<{!>,<ou!!!>>,{<!!!!{"!>,<!a!!>}},{<{",!>u>,{}},{{<i!!{oe!>,<!!!>,{ee<>,{<!i'!>!!a'>}}}},{{{<!{<!"eaa!!}''<}uu{u>,{}},{{}}},{{<i,!!}{a!u!!!>>},{<}'!!{!!o{!a!!{"!>a>}},{{<!>a!>,<!>e!>,<e!>u!!!!!>o{>}}}}},{{{{<a!,i!>},<uee!!!>i>}},{{<!!!!!>>,{{},{<!!!>!!!!!!o{!!!><e!>!!aa}e!e<,"i,>}}}}},{{<o'!><i!!!>!"!!!>>,{<}a!>},<!,>}},{<'u!>e!!>,{{},{}}}},{{{<}!'"{i!>},<>},{{<!!,'ai!!!!!!!!!!o'!>},<o!a>},{<!><"<!!!>!"o'!!!>!!e!!o>,<a!>,<<u,}e!!!>!a'!ii"!">}}},{{{<"!u!!u!a!>},<!!!>,<{>}},{<}>}}}},{{{<<o!'!!!>},<!>,<!>,<!>},<iu!!!>!><e!a"">},{{{<!!''!!!>},<,,>}},{<'',!>},<'"!>ii!>},<<!!<!!!>!!i>}}},{{<e!>!>},<}!!!>u!o!!o>,{<<,!>,<o"!>!>{!i>}},{<ie!!a!!}!!e"}{!!!>}!>!!!>,,!!!>e>,{}},{<>,{<ou"!!!>e!>,<!>},<ae!!u!!'>}}}},{{{{{<{,""a!!ai>},<!>!!!>!>o,!!}!>},<}!>,<ia!>ua>}},{{<!>,<!>}"e,}>,<!}!>,<u>},{{{{<e!>!!!>},<!!a!>>}}}},{}},{{{<!e!>"io,oaii!!!!!!u!>{<u,>}},{<a"!!}!>!!!>a<e'u!>,!!!>"o!>,<i!!>},{<!!ea}!>a!!u!!>}},{{<,<i!o>,<'aa!,!>,<!>,<>},{<!!!>},<u!>},<{!!!!!>!>},<!}iie!"!!o!!!>,!'">}}},{{{{<i!"!!!!!>,<!>!>,<!{!>!{!!!>e!!!!a'!>},<!!!>!>,<>,{{{<!!!>},<ae!!io!!a!!>,{<!!!><!!!!io!!{u}o!!!>,<">}},<,!o,e!!}!}!>},<!{!!o!>!!!>",>},{<iuoo!>'o!!!!},o!<!>},<!>!}!>,<>}}},{<"{"!u!!!>!!!>,<u!>,<!>>,{<{}!,<e!>,<!!e{!!!>>}},{<u{!>,<iu!>,<u!!u{o!>},<>,{}}},{{{<{!!!,',,!>a>,{<"!!!>,<!>ua!!e!!ia!><<ao!!>}}}}},{{{}},{{}},{}},{{{{{{{}},{<!><a!!!>!>},<uu!!a'uu{!>,<}!>,<'>}}},<!>iua>},{{<!!u}!!!>},<o!!!!a<uo,!!!>>},<!>u'e{!}">},{{<!!!!!>>}}}}}}}},{{{{{{{{}}},{{},{}},{{<!>},<uo>}}},{{<!!'!!!!!>,<>},<'ueuui!>,<,>},{{<"o<!>},<u,!!!>},<!!'o}u!u!>,<!!>,{<!>>}},{<"!!!>>}}},{},{{<",i">}}},{{<,!!!ue!!a,!>>,{<o<!!i}!>},<{>}},{{{},<o}u!!<!!!>",,i"!"!>!"'"!!o!>},<>}},{{{{<{!!!>},<!!a<!!!!i"!!!>>}},{<{>,{}}},{<a!!!>},<<ie!!!!>,{}}}}},{{{{},{<oo!!u!!!>o<!>,<""i!!!i!>!>,<!>,<!<,}<>,{<!,u<>,{<!!i'!aui!!o>}}}},{{{<!>,<!>,<ae!!!ia!>i<!!!o!>,<}{a!!,i">,{<>,{{<!!!>,<o!!a"!}o,!>,<,o'!>},<!>,<!>},<>}}}},{<}!>},<!!o}>},{<e!>!!!u}!>},<!>},<u!!,'!!!>>}},{{{{{<i,<!>,<!}'"!!{!!}>}},<e}!>},<{}!!>}},{<!>},<!}}a!!e!!u<!>>}},{{}}},{{{{<,!>},<!!!!!!!!{>},<{!>,<!>!!}!!!>,<!!!>}!!!!o>},<eo""<i}aie!<!>,<!>>}}},{{{<{o,!>},<>},{{<<>},<ao"a>}},{<!!!>'!!!><o!>!<!>!!!!"'!"!!!!!,!>},<,!!u>,{<!o'e!>!>},<!!i!>!!}ii>}},{{<<!}{u!!>}}},{},{{{},{<,!!i!!,u!>},<i}>}},{{<!>!!!>},<e!u}!!u!!!>'<o}}u!i<!!!!u>},<>},{<!u,>}}},{{},{{{<"!!!>a!!i!>{!!e'<,i!!"'>,<}!!!>!>},<!!!!<!>{"!!o">},{{}},{{<eo{"uuo}{!'!u!}>}}},{{{<!"'u'i,>},<'!!a{!}!e'""!>">}},{{},{{<<ae!>},<i!!u!e"!!!>{a!!a!!!>>,<!>,<io!>},<!{>},{}},{{{{{<!!!>!!i',,{a!!!>ue!!!!a>},<o'!!!!!>!!!,!>,<>}}},{},{}}},{{<!>},<!>,<!!!>!!}{>,<!!!>">},{{},<!e,!!!>},<!e!!!>eu"!!!>u}>}}},{{{{<>},<"u!!!!!!!{!!<!!o}<!>,<{,!>,<}!>,<o!>!">},{{{}},<!<'e<!!!>"i!!!>!>},<u!<{'!>},<!>},<>},{{}}},{{{{{},{{<o!>},<a<'!>e!>,<,">},{{{<o!!'<u"!!<!o,!>,<!>},<"!">},{<u!!!i!!u}o!!!>a!!!>u>}}}}},<!!!}<!!<oo!!!>,<!!!>ee!!!!{e"!>!>,"!!>},{{{}},{}},{{<<!><,}{,i}eai!>"e}>},{}}},{{{},{{{<>,{}},<!!!>,'o!>,<"oi"!!!>,!'>},{},{<u!>'!!<>,{<e!!!>,<!!!>a>}}},{},{{{}},{{{<!!!e!a,">}}},{{{<>}},<!!,}"e!>},<!!!!!>!!!>a!!!>!>{!!i,!!!"!>},<!<>}}},{{<!>},<!'"!!!>",ie'}!o>},{<>,{<!!!!!>!>},<<'!!}"}{o,u'u!!<!>,<'>}},{{{}}}},{{{<}e!>!!!>u>},<!!!!u!!<!>},<"!>,<""<!!}}!!!!>},{{<!!o!>,<!!o!!e,{e!!!u}!>},<i>,{}},<!!!>},<{!!}!!!!{!,i!>},<>}},{{{<!>},<!>,<!!"'<!aa<!>o>},<!<'o<>},{{},<!eo!!a<{'o!,<{}',!!}e!>i>},{{{<"!!!>o!>},<!ua!>{<i!>,<!>'i>}},{}}}},{{{{}},{{<!!!>,<"u!!e!>},<!!!o!>},<ia!>,<'<>}},{}},{},{{{}},{{{{<!!ia!!ou{e>},<!ua{!!!>>},{{{{{<!>},<!!'!!}!>,<u!!!!oi!!a>}},{<"!>!>,<{<e!>},<ii!!!>},<i!}!>,<'>}}}},{}}}},{{{{<e!>},<!>"}oee<'<e"{,!}<o!>>},<e!!!!i!>,<!!i!!!>"!>},<e',{!!!>eoue>},{<}"{>},{<!!!!i{oo"{{"!!","u,!>'>,{<!!,>}}},{{{}},{{<<ue!}u!>'!<"!!!!!>!>!{!>,<u!!u!!!!!>>,{<!'a!e!!!>}u!!!>},<{!'u>}},{{{}}},{<}a!!o,!>,<o!!!!!!!!au>,{}}},{{{{<i}<!>,<a!e!!eaa{!>,<<!!>},{<!>,<aii!!!ia}e>,<}!!!>oeau{i}!>},<i!!<o!!!{}>}},{{<,,,!>,<!><!!iu"ie!">}},{{{<!>,<!>e,e!>},<!!u}!!!>'!>,<>},<{>}}},{{}},{{{{<,!!'{!>,<!>},<!!}!>,<a!i!!!!<!!!!'a{>}},{}},{{<}}!uee}!!!!!>{e{!>},<ou!>},<,>},{}}},{{{{{<!!!!!>a,a!u!>u>},{}}},{}},{{{<i''!>,<,"!{>,{{{<o!<!!{!!!>!>e{!>,<<e!'a>},{}}}}},{{{}}}},{{{},<,"!>},<{>}}}}}}}},{{<!!!>!!!>'e<!{,uieu!>},<e>},{<!>},<,!>,<!>,<!!!>!>a>}},{{{{{<i<!>''!>,<}}!!!!!!!!!!!>!!!>!>,<'>},<!!'!!}i!>,<!!!!!>"oi"!!!><<ai>}}}}},{{{{<!>,<e!!!>"<<!!}!!!>!!!i!!!!'!!!>!!!>>},{{}}}},{{},<o!!!>"!!!>!>},<,>}}}},{{{<!!!>u!!!>!!""o{,a!>,<<>,{}}},{{{{{{{<,"!>},<>},<'!!,!!!>{e!!!o,>},<!!!>!>,<'<!>},<'{}!!!>!>,i!!!>"!>,<{>},{<>,{}},{{<!!,!!!>>,{}},{<>}}},{{<"a{{!o!>,<!!!>!>,<o!>>,{<!!!e!><}>}}}},{{{<a!!}u!ie!!!>!!!>!!,!>},<!!!>!!!!!>u{!u>},{<i!!'>}},{}}},{{{<"}a'!!{!!ua}>,{<}!!!>aa'!!<!!!!!>!,,{ai!!!>'"a!!!>'!!!>>}},{},{{{},{<ee!!>}}}},{{{}},{<<i!>},<!'!<!!!!!>!!a!!au>},{}},{{<!iu!>oe{a>},{<e""{!>,<'!>},<,{ieaa>},{{<a!>,a!!u!>},<!>!">},<a!o!!!>,<!!!>{!e<!>,<>}},{{<i!!'!!!!>,<,<!!,{<ii>},{{<o!!!!!>ui!>},<!>,<e,<a!>!>,<!>},<{>,<!>},<}{oi!>},<i!!"u!!a{!>,<<",!>},<!!'u>},{{{{},{<},}!>!>,<>}}},{<!>!ua!!}{uue{!!!!!>!u!!!""!!}!>,<'!>,<>}},{{{<!>},<!>},<!!'<!!<>,{<o!e"'"!!{!!!!!>,<>,{<!!aoai,>}}}},{{},<,oe<!!iueo"u,!>}!!a>},{{},<u,}!!!!!!a{<!>,<a!"'!!!i!!'>}}},{{},{{<i{,"i{>},{{<!!!>!!!>{!>,<{!!{u!!!>!>,<!!!>>},<{{!o""oo!!!>>}}}}},{{{},{{{<!!i"!!!>!>,<!i>},<}!!!>},<!!o{>},{<<ue{!<u<!!!><!!!>a!>,<>,{<e'!a,a!!!!<aai>}}},{<a!>"!>o}!i!!!>''o!>},<'>,{<!!!>!!!>,<}i>}}},{{{{},{{<!!">}}},{<eou!>,<<!>,<!!!><!!!!!>},<>,<u!>},<a,!!<<e!!!>,<!>},<oiu>}},{{{{<!>!!!>!!i>,{}},{<,!!!>,<!!"o,i"!>},<{!>},<!><uu!>!>e>,{{<!i<a!>,<}aae{!>,<'<>}}}},{},{{<i}!>},<}!"!>},<!>>},{<e}}!>},<e<>}}},{{{<!!u!!ae!!!>},<!>},<!<i!>{"!>!>o<>,{{<!!!!!>,<{!!!>!!a!>!>},<!>,<{<<!>},<e<"'>},{}}},{<i!>}{u!>,<o<!>,<!>},<<!!!!!!!"e!!,a!!!>,<>}},{{<e'o!>},<}!!a!!}!!<!!!!!>!!u}!>o>,{}},{<!>'!>,<!!!!!>},<>}},{<!!!>!>,<!ua!}}!oe!!!'!>!>,<,!<e>,<>}},{<"}!i<!!!!!!!>},<<!!o!>},<'>}},{{<>}}},{{{<'}{!'u!!!>},<!!,!!!>!>},<a{">,{}},{<"!!}!>!!!>,<!>'!!!>!!!>},<>}},{<>,{<,}a!>{a<,,!>},<"!!o{!>,<a!>!!!!!!i>}},{<a,"<}!!<{!>,<{>,<!!!>,<}}!!o!!!a!>},<a!o!!!!!>},<!>},<!>},<!!!>,<>}}},{{{<!!{!!{!>},<u}u!!{e!>!>!>e,}!!!>},<!!<,}>,<!!!,u!!!!,uo!!!>ai!>u<u!o!o!!e>}}}}},{{{<}!}ue<{'a!!u>,<!!i!>},<'!!!>},<!!!>,<!>!!!>!e''{<!!!>eo>},{{{<uoo"{e!>,<!>,<!>!!}!>!ao!>,<<,e!!!!>},{<'o''e'o!o{!!!!!>!>!!u!!>},{{}}}},{{<uie,oa"i'o,!>,<'>},<"!a{!!!>>}}}},{{{},{{<e!!!>}o{oi!!,>},<'{o!!!>,'i"o!o"a,a!>,<!!!!!>e!o>}},{{{{},{<e!>}}!!o{!>,<!>!>!!!>e!>,<!>},<"u!>,<{u>}},{{},<{o<u!!!>},<<{!'!!!>!>},<oe!!!>!">},{<!>!!>}},{{{<a,!o!{{!>},<>},{<}!>},<i!>!!!>!!"o>}},{{{<u!!!>!!<!>,<!>u'<!>},<"!!!>!<!!!>}!!!>,>}}},{{<u!!!>,<{ou>},{{<<!!"i!<!!e!>},<",'>},<'!>!!!>!o!>!>!!{!>!!e<!e>}}}},{{{{<!!!!!!!>},<u!>,<}!>},<!!i!<!>},<!><!!o!!<<{!!!!>,{<}'!"i{a!>},<!!!>!>},<!!a!>,<o>,{<!>i!>,<'!>,<"!!!!!>>}}}},{{<ao!>!>!>},<!"!>,<<!!!>,o!!u>,<{>},{{<!!!!!>>}}}},{{{},{{{<a<'eaau!>!!'i<!e!>,<!>,<>}}},{{<!u!<,a<e!>{a!>o}!!}"!'!!>},<eu!!u}!!,!!!>},<!!"!i}>}},{},{{{{<!>!>,<e>}},{{{<<!!!!!!!!!>'!>,!>">,{}},<!>,<!>!!!>>},{{{<ooaoe,u>,<i>},{<}!!!>!u,!>},<'!!uoo>}},{{<!!!>!o>},{<!>,<!>,<i{'>}},{<!!!>!>,<!>,<{!uuo>,<aae"<",>}}}}},{{<!!!>,<!!!>o}iu>,{<i}!!!>o!>},<'i,'>}},{},{<!>},<i}!>,<'>,<>}}}},{{{<!!i!>},<}!!{i!o!!!>,<!>},<i{!!<!!i!!i{>},{<>}}}},{{{{{{<>},{}},{<>,{<'>}}},{{},{{<i!!,{a!i}a!e<{!!e!oi<o!>},<}>}}},{}}}},{{{{{}},{<"o{!>,<>}},{<!>,<{"!o'!!!>!>,<o<'}uii!!,!!{<!>},<!>},<>,{<o!!a>}}},{{{}},{{<>,{<oa!!!!}!>{!>'"<!,<!!!>,<!>},<a!>!>!!'>}}}},{{{<}"}!>},<,"!!!>},<!!!!!>,!>'u"<!>},<u!!!!o!!!>},<>},{{{<}!oii,}!>!<"}<a!!!>!}!>,<>}}}},{<!>!!!!!!!>oi!>oe!>},<>},{{}}}}},{{{{<uu,o!!!>u'!>a!>aoa!>},<>},<!!!}ue!>e{{!>},<a!{i!!uo,!>},<>},{{<}!!a}o!>}!>,<!u!>!>},<,ueaoe!>,<!!!>>}}},{{{}},{{<!>,<e"!>>,{<!!,!{!!au}!!"u!!!>"!>,<!!>}},{<"!!,!>}!>{oe!>},<uu!!"!>},<>}},{<>}},{{{<"!>},<>},{},{}},{{{<!!<i{{e!>,<!!!>"}!}e!>,<'>}},{},{{{<,ee>}},{<'e!!!>e!<!!!!!>,!!!>!>},<{}!!!!!!<>,{<!><!>,<a>}}}},{{<!!'eu}{>},{{<ai!>,<e!!!!!>!!!>},<o>},<!>,!>},<!>},<u>}}},{{{<!!!!e"<<!!!>},<}!!!>!>,!!{"'!>,<{!>},<{o>},{{{<!!!>!!!>{"!!!!!>,{!>,<o!>e<!>,<>},{{{}}}}},{<}a!!o!!}!'!{!!e!!oa!<!!">}}}}},{{{{<'uu!>},<',!,!}!>,<!!!>},<>},{{<i!!!!!o!!!>i!!{>}}},{{{}},{<au,o,a!>,<"!}ai!>o>,{<!!!!!}!>},<<!!u,!>!!!>!!}o!!'!>'}!!!>!>,<>}}}},{{},{{{<""a!!{!>,<!!!!!!ue!!uui!!<!>},<>},<,!>},<i!>},<a'!>,<{o!a"!>},<!!'!!!>!,<{>},{{<e}<!>,<a!!'!!!!!>e!>!!a{,!!!ui!!o>}}},{{{<e!>!>,<u",e!!!!i"e>,{<o!!aia}!>!!!>i,!!!<ie>}},{{<{!>,<>},<u,<a!!!>au!>,<ie'!>,<'i!!!>>},{{{{{},{}},{<!>},<<!>,<e!'"!>,u}{>}},{{},{<"u!!e{,!uu}!!}!!!!!>!>},<iue}!>!!!!!>>}}},{{<!!}!!!>!>!>},<,!>'i,e!!!a'!!!}>},{}}}},{<{<<!><'!oi!!e"!>,<'oi}>,{}},{{<,!>},<!!o!>,}io'!>},<i!>},<{!a!!!>,u>},{<a<!!!>!!!!!>{i!!!>,<i!!u<!>i"{!>},<>}}}},{{{<a!!!>,<>},{{{},{{}}}},{<!!">,<!!!!!!e!!}!}{!>,<<!!!>>}},{{<o!>,<'!>!>,<!>},<e<!!>,{{}}}},{{},{}}},{{},{{<!}!>},<ioo"!>,<'!!>},{<!>,<e!}>}},{}}},{{{{{},{{<!!'e',,a!>},<!>!!<}{!>},<<>,<<oa!!i>},{<!>,<!!u!>},<!>,<!!{!>,<o!>},<">,<>}}},{{{<!>!>},<<o!!aa<!'!',u{!!i<!>},<e>,{{<!!a"{{'!!a!!!!!>>}}},{},{{{{}}}}},{{<!!!'!>},<oou!!!eu!o}e!>},<!e{>},{{<!>,,!<i"!>i<u'!>},<!>!>},<!!!>}>},{{<'!!"{u!ia{{!!!>!!!><{!!!>!!'>},{<!!!>!!!>}{}!>,<<!ae},!>i!>,<>}}}}}},{{{<<!!!!!>!!eu!!e!!o!>',i!!!>},<",!!!>},<,o!>,<>,{<},}!!{!!!>!>},<'!!'o!>,<>}},{{}}},{{<{eo!!!>'!!!>'>}}},{}},{{{<!>,<ee!!oi}!!!!!ee{<!!!>u!!!>!>},<">},{<o}!!!>"!!!>}!>">,<}<"!',,<{!>,<!!!!!>,<{}>}},{<!i{,e<'!!'!!iau!!!>>,{<!!!>},<u}o!>},<!!!>'!!o'!!i!>},<!!!>!a>}}},{{},{{{<!!"<"!au!>eoi'!<e!o!!,!!io>},<<!'!>!>},<!!!!uaa!>,<o!{i!>,<'ie{a!!!>>},<!>!u,o{}iuu!!!>}!!!!!>{o!>>},{{{}}}}}},{{{{{{{<!!,>},{{<"!!!!!>u!e>,{<>}},{{{<'>}},<!!,!!!>,<{ia!!!!u{>}}},{<}ia!!!>!>},<!>,<!>},<'"!!"u!o,!!!>},!>>},{}},{{<}!>,<!>,<u!>},<'>,{<{u}}!!!>,<!>,<oa!>!>},<i!>},<<>}},{{},{{<{">}}},{<e!>,<,!!}iu',e{'!!!!!>}iu>}},{{{<,{!>},<!i!>,<}!>},<}!!"!!,e!!!>!>i!!!!!><{>}}}}},{{{{{<!e!!!"!>!,!!!>},<,!!<!!!!"u>}},{{{<!{ua<">,<"<<!!!>!!a!>{!!o!!!>>}},{{}}}}},{{{<!>,<},>,{<'!!!>!!{i<i!>},<i}!>,<!>},<iu>}},{<!!{!!!!!!!>e!!!>'i!>,<!!!>!!"<ao'u'a!!>,{<,,!!!>,<!!'!>>}},{{{{{{<!>},<>}},<!aei!,e!!!>}!!u!>},<eui{}!o!!i'}>}},{{}}},{{},{{<!>},<'<!!!>},<>,<!!'}}u}!>,<>}},{}}}},{{{}},{{<!>,<!!!!a,!>,<au>}}},{{{{<ui{u<!!e!!,}>,{{<!>o!>},<!>'<!>"}!!!>e,"!!a>,{{<'i"!!!>!!!>!!'!!!>!>'!>!>},<!!e'{!!,<>},{<!{!!!!!>}!!!>},<!!!>!>},<ou,,!!a<!>,<!>},<!!!>,<>}}}}},{{<!>},<<{i!>},<}'e'ee<!!!>!!u{{o!!!>'>},{<!!!>!<!!!>a!>,<!!!>!!o!!!!!>>}}}},{{{{{{{<"!!!!'oo!!!a}!>,<{{e!!'{>}},{{<!!"}',!!i!>!!"}!<<iu!>>}}}}}}}}},{{{{},<}i'>},{<<!!!>},<!!!>a,}'u!>!!!><'{,ee<!>a>},{{<!!!!<eo"}}e!,!!!!"!!!>>,<!>,<!!!!ieuu!>{>},<,o,>}},{{{{<e!i!!!!'ea',iau,!!{i!!<>,{<!!!>ue>}},{<>}},{}},{{<!!}<!a!e!!!!u!>'{o!!!>,<{}aeuu>,<!!!>e,!>},<!!!>},<!>!a!!!!io{ui!>>},{<!!!!!>!!!>'"!',i!>,<'!!!>!>"}!>!!,!!o>}},{<!}!>!>,<}!!!>,<"ooe!!o<"!!!!">,{{<a!!!>,<<!>,<!>},<!!!>"!>},<'!>,<i!>,<eoa>},{<<!!",ia}"!!{!>},<<!"!>},<!>ou>}}}}},{}}},{{{<>,<a!!i!!!!eeeo>},{<<!">},{{<!ii">,{<,,<u"!e!>,<,}!>},<<<!o!!,}!>},<>}},{{<!!uo"<<>}}}},{{<!>},<<!{u!!!>},<{oei}o!i!e"!'!>,<!>'>,{<a{>}},{{{}},{{<ao{{!!">}}}},{{{{},{{{{<!>!!!>!>e'e>}}},{<o,,!!!>!!!>!>},<i'<>}}},<ii!"!!!>a!>'!>},<e!,>},{{{<!!!!!'a!>,<<i>}}}}},{},{{{{{<!!!>!>},<!!!>aoi!!!!!!!>!>},<!!!>>}},{{<!>!>,<a!{!!!eo!!,!!i,}!>!>}",!!!>!>,<!>>,<"!!!!<>},{<!>,<!>},<io!>},<u!>,<'i!,u!>,<>}},{<}ea,!"<e!!!>},<>,<,e>}},{{<}!'!!<>,{<!>},<<!>,<!!!!{a!>u}!>,<!!!>'!!!>ia,!!>}}},{{{<!{!>uou!!''',!>u!!!>,uu!>o{!>,<>},{}},{{{<!!!!!>oau!{''i!i!!ii>}}}}},{{{<>,{{<,e!!i!>!>,<{>}}}},{{{<,<eu!>,<}!!"o!!!oe""!>}!!!!<"'>,<e!>},<!!u!!!>,<!>},<!>,<''!!!!!>,!!!!,!>,<">},<{!!!<!!{e{!!!>!!!!i<!>!>},<i!!!>!a!!,'e!!}!>,<>},{{<'!!<i}"}ou<!!!>'!!!>!>},<!!u!>>},{{<!<!!}a<!ooo>}}}},{{{{<",>,{}}},{<{!>,<!!!>!>,<o!!!>"'!e,ei!}!!i'>}},{{<,'!o>},<,e'>}}},{{},{{{<e!"!>"aa',!!"!!'>}}},{}},{{{<!}<!!!>!>!>},<"''<!!!!a!!!<!!'>,<!>,<}">},{{{<>,{<o,{{}!>,<e,!!!>i,!>ui!!!!!!!>!>},<{e>}}},{{<,>},<,!}!!!>'"!>!>!>},<,!!i",o}i>}},{<u!!}>,{<!!!>!!<a'}e!!oe!,!>,<!e"io<}!!!>>,{<!!!>u'a"!>,<{"<e{}!>>}}}},{{<!>,ie"}!>},<u,!!u'e>,{<"!>,<}!>,<u>}},{<e}"!>},<!!!ei!!!><>,<}!!!>!>,<>}},{{<iu,!!!>!!"}!!!>!!!>!!{!!!>ua>,{<!>,<<}>}}}}}}}`
  output = stream.groups_score(input)
  console.log(output)
  assert.equal(output, 17537)

  output = stream.garbage_count(input)
  console.log(output)
  assert.equal(output, 7539)
}